src/kademlia/traversal_algorithm.cpp

```diff
@@ -118,85 +140,110 @@ void traversal_algorithm::add_entry(node_id const& id
 		done();
 		return;
 	}
+
+	o->flags |= flags;
+
 	if (id.is_all_zeros())
```

Comment:
the observation I made was that bootstrap nodes (i.e. the ones we load from our state file) never have node IDs associated with them. it's unnecessary to generate bogus node IDs and then sort the nodes based on them. Instead, allow the tail of ``m_results`` to be unsorted, and function as back-up nodes when lookups fails (and don't yield more nodes)